internal/generate/collector/manifests.go

```diff
@@ -212,6 +212,7 @@ func (c *Manifests) addRoles(rawManifests ...[]byte) error {
 		if err := yaml.Unmarshal(rawManifest, &role); err != nil {
 			return err
 		}
+		role.SetNamespace("")
```

Comment:
It would be easier to just remove an object's namespace before [writing it](https://github.com/operator-framework/operator-sdk/blob/master/internal/cmd/operator-sdk/generate/internal/genutil.go#L90).

Reply:
Hmm, the package and naming of `WriteObjectsToFiles` makes me hesitate a little putting this logic there since it seems like a very general-purpose and obvious function. I interpret it as "please serialize these objects directly to files".

WDYT of removing the namespace of all objects just before they're passed into this function and only when we're using it in the context of writing a manifests directory?

Reply:
Actually, how about [here](https://github.com/operator-framework/operator-sdk/blob/497b7f8ce68cd4d8f1fdd4a70f9cd78f80149654/internal/cmd/operator-sdk/generate/internal/manifests.go#L23-L45)?

Reply:
:+1: to your last comment.